### PR TITLE
Now you can invite people who aren't in the system yet by e-mail

### PR DESCRIPTION
**Pull request by amandasavluchinske**
They can choose whether to sign up manually or with Google and will be redirected to a battle with who invited them once they sign up.

### CODE REVIEW

**Comment by chocoelho** [comment]
Should I have any key to check this out?

File: accounts/helpers.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+            challenger=invite.inviter, challenged=user)
+        Team.objects.create(trainer=battle.challenger, battle=battle)
+        Team.objects.create(trainer=battle.challenged, battle=battle)
+
+
+def associate_invitation(backend, user, response, *args, **kwargs):  # noqa
+    invite = Invite.objects.filter(invited=user.email).first()
+    if invite:
+        create_battle_by_user_invite(invite=invite, user=user)
+    return
```
Comment (chocoelho) [inline accounts/helpers.py]: IMHO, it looks more pythonic to go like this:
```python
if not invite:
    return
create_battle_by_user_invite(invite=invite, user=user)
```

File: accounts/views.py [hunk 2]
```diff
@@ -35,7 +38,22 @@ def form_valid(self, form):
...
+    def get_redirect_url(self, *args, **kwargs):
+        invite = Invite.objects.filter(invited=self.request.user.email).first()
+        if invite:
+            if not invite.has_accepted_invite:
+                create_battle_by_user_invite(invite=invite, user=self.request.user)
+                battle = Battle.objects.get(challenger=invite.inviter, challenged=self.request.user)
+                invite.has_accepted_invite = True
+                invite.save()
+                return reverse('battles:choose_team', kwargs={'pk': battle.pk})
+        return reverse('battles:new_battle')
```
Comment (chocoelho) [inline accounts/views.py]: IMHO, it looks more pythonic to go like this:
```python
if invite and not invite.has_accepted_invite:
    create_battle_by_user_invite(invite=invite, user=self.request.user)
    battle = Battle.objects.get(challenger=invite.inviter, challenged=self.request.user)
    invite.has_accepted_invite = True
    invite.save()
    return reverse('battles:choose_team', kwargs={'pk': battle.pk})
return reverse('battles:new_battle')
```

**Comment by amandasavluchinske** [comment]
@chocoelho should be ok now :) In case CI fails don't be scared, I'm having an issue with a lib which is using two different versions of another lib and that's breaking CI :/

**Pull request merged**